tests/functional/simple_seed/test_seed.py

```diff
@@ -1,19 +1,290 @@
+    seeds__disabled_in_config,
+    seeds__enabled_in_config,
+    seeds__tricky,
+    seeds__wont_parse,
+)
 
 # from `test/integration/test_simple_seed`, test_postgres_simple_seed
 
 
-@pytest.fixture(scope="class")
-def project_config_update():
-    return {"seeds": {"quote_columns": False}}
+class SeedConfigBase(object):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "seeds": {
+                "quote_columns": False,
+            },
+        }
+
+
+class SeedTestBase(SeedConfigBase):
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self, project):
+        """Create table for ensuring seeds and models used in tests build correctly"""
+        project.run_sql_file(project.test_data_dir / Path("seed_expected.sql"))
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        seed_actual_csv = read_file(dir_path, "seeds", "seed_actual.csv")
+        return {"seed_actual.csv": seed_actual_csv}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "models__downstream_from_seed_actual.sql": models__downstream_from_seed_actual,
+        }
+
+    def _build_relations_for_test(self, project):
+        """The testing environment needs seeds and models to interact with"""
+        seed_result = run_dbt(["seed"])
+        assert len(seed_result) == 1
+        check_relations_equal(project.adapter, ["seed_expected", "seed_actual"])
+
+        run_result = run_dbt()
+        assert len(run_result) == 1
+        check_relations_equal(
+            project.adapter, ["models__downstream_from_seed_actual", "seed_expected"]
+        )
+
+    def _check_relation_end_state(self, run_result, project, exists: bool):
+        assert len(run_result) == 1
+        check_relations_equal(project.adapter, ["seed_actual", "seed_expected"])
+        if exists:
+            check_table_does_exist(project.adapter, "models__downstream_from_seed_actual")
+        else:
+            check_table_does_not_exist(project.adapter, "models__downstream_from_seed_actual")
+
+
+class TestBasicSeedTests(SeedTestBase):
+    def test_postgres_simple_seed(self, project):
+        """Build models and observe that run truncates a seed and re-inserts rows"""
+        self._build_relations_for_test(project)
+        self._check_relation_end_state(run_result=run_dbt(["seed"]), project=project, exists=True)
+
+    def test_postgres_simple_seed_full_refresh_flag(self, project):
+        """Drop the seed_actual table and re-create. Verifies correct behavior by the absence of the
+        model which depends on seed_actual."""
+        self._build_relations_for_test(project)
+        self._check_relation_end_state(
+            run_result=run_dbt(["seed", "--full-refresh"]), project=project, exists=False
+        )
+
+
+class TestSeedConfigFullRefreshOn(SeedTestBase):
```

Comment:
Just a reminder that if these tests are moved into the adapter zone, they'll all have to have names that don't start with 'Test' and a simple class wrapper that does start with 'Test'. Doesn't have to be done now though.